pwndbg/gdblib/memory.py

```diff
@@ -45,7 +47,11 @@ def read(addr: int, count: int, partial: bool = False) -> bytearray:
         if not hasattr(e, "message"):
             e.message = str(e)
 
-        stop_addr = int(e.message.split()[-1], 0)
+        stop_addr = addr
+        match = re.search(r"Memory at address (\w+) unavailable\.", e.message)
+        if match:
```

Comment:
I think this should be:
```python
match = re.search(..)
if match:
    stop_addr = ...
else:
    stop_addr = int(e.message.split()[-1], 0)
```

to preserve the old behavior of taking the stop address from any msg

Reply:
I think this might be trying to catch TARGET_XFER_E_IO in https://github.com/bminor/binutils-gdb/blob/70fd94b244573225cb04ae41101d495def78b9e6/gdb/corefile.c#L169-L173, I just happened to trigger the other case (TARGET_XFER_UNAVAILABLE). I could turn this into an explicit regex, I suppose?